cmake/modules/CheckDependentLibraries.cmake

```diff
@@ -177,46 +177,18 @@ if(WITH_FFTW)
   endif()
 endif()
 
-if(WIN32)
-  set(BLA_PREFER_PKGCONFIG ON)
-  set(BLA_PKGCONFIG_BLAS "openblas")
-  set(BLA_PKGCONFIG_LAPACK "openblas")
-else()
-  set(BLA_PKGCONFIG_BLAS "blas-netlib")
-  set(BLA_PKGCONFIG_LAPACK "lapacke")
-endif()
-
 if(WITH_CBLAS)
-  # find_package(CBLAS CONFIG REQUIRED)
-  pkg_check_modules(CBLAS QUIET ${BLA_PKGCONFIG_BLAS})
-  if (CBLAS_FOUND)
-    add_library(CBLAS INTERFACE IMPORTED GLOBAL)
-    set_property(TARGET CBLAS PROPERTY INTERFACE_LINK_LIBRARIES
-                                      ${CBLAS_LIBRARIES})
-    set_property(TARGET CBLAS PROPERTY INTERFACE_LINK_DIRECTORIES
-                                      ${CBLAS_LIBRARY_DIRS})
-    set_property(TARGET CBLAS PROPERTY INTERFACE_INCLUDE_DIRECTORIES
-                                      ${CBLAS_INCLUDEDIR})
+  if(NOT CBLAS_PREFER_PKGCONFIG)
+    set(CBLAS_PREFER_PKGCONFIG ON)
   endif()
-  include(FindPackageHandleStandardArgs)
-  find_package_handle_standard_args(CBLAS REQUIRED_VARS CBLAS_LIBRARIES
-                                    CBLAS_INCLUDEDIR)
+  find_package(CBLAS REQUIRED)
 endif()
 
 if(WITH_LAPACKE)
-  # find_package(LAPACKE CONFIG REQUIRED)
-  pkg_check_modules(LAPACKE QUIET ${BLA_PKGCONFIG_LAPACK})
-  if(LAPACKE_FOUND)
-    add_library(LAPACKE INTERFACE IMPORTED GLOBAL)
-    set_property(TARGET LAPACKE PROPERTY INTERFACE_LINK_LIBRARIES
-                                        ${LAPACKE_LIBRARIES})
-    set_property(TARGET LAPACKE PROPERTY INTERFACE_LINK_DIRECTORIES
-                                        ${LAPACKE_LIBRARY_DIRS})
-    set_property(TARGET LAPACKE PROPERTY INTERFACE_INCLUDE_DIRECTORIES
-                                        ${LAPACKE_INCLUDEDIR})
+  if(NOT WITH_CBLAS)
+    message(FATAL_ERROR "LAPACKE support requires CBLAS")
   endif()
```

Comment:
Is cmake_dependent_option appropriate for this kind of logic ? https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html#module:CMakeDependentOption

It might be at the place of defining the option, this validation is still fine here I think.

Reply:
With following code:

```cmake
cmake_dependent_option(WITH_LAPACKE "Build with LAPACKE support" ON "WITH_CBLAS" OFF)
```

... and configuring with `-DWITH_LAPACKE=ON -DWITH_CBLAS=OFF` would silently just ignore LAPACK, which I don't think is particularly helpful. Better to fail hard.

Reply:
I was afraid that it would be like that. In that case, your solution is appropriate!